docs/enhancements/status-reporting.md

```diff
@@ -0,0 +1,117 @@
+# Status Reporting
+
+<!-- markdown-toc start - Don't edit this section. Run M-x markdown-toc-refresh-toc -->
+**Table of Contents**
+
+- [Status Reporting](#status-reporting)
+    - [Summary](#summary)
+    - [Objective](#objective)
+        - [Goals](#goals)
+        - [Non-goals](#non-goals)
+    - [Proposal](#proposal)
+        - [For Sonobuoy Aggregator](#for-sonobuoy-aggregator)
+        - [For `sonobuoy status`](#for-sonobuoy-status)
+        - [Heartbeats (optional/future)](#heartbeats-optionalfuture)
+        - [User Stories](#user-stories)
+    - [Unresolved Questions](#unresolved-questions)
+
+<!-- markdown-toc end -->
+
+## Summary
+
+Sonobuoy -- both `sonobuoy run` and through Scanner -- has really lackluster UX.
+At present the tests will be kicked off and no information will be reported
+until the run either succeeds or times out. This is not ideal. We would like to
+have a concept of test status that can be reported, in near-real time, back to
+the user running the service. Both the CLI and Scanner should be able to display
+this progress in a way that makes sense for their individual UX.
+
+## Objective
+
+`sonobuoy run` and Scanner should report the status of an ongoing Sonobuoy test suite.
+This status should be secure so only the intended audience can see it.
+
+### Goals
+
+* Create a command `sonobuoy status` that shows the results of an ongoing run.
+* (optional) Workers can report their status to the aggregator
+
+### Non-goals
+
+* Actually updating Scanner is outside of our scope, though we should make sure
+  the API is useful to that team
+* Complex metrics and monitoring. The first pass should be simple.
+* Scanner support
+
+## Proposal
+
+### For Sonobuoy Aggregator
+
+An annotation will be attached to the sonobuoy pod, periodically updated by the
```

Comment:
What sorts of status will this provide that we don't have already? Our aggregation server already knows if stuff is running (is it waiting for results or not). I am unsure how this this proposal is going to improve the UX for scanner when we already tell the user it's running and then report failure after a timeout.

If this is an infrastructural phase, what do you think about writing out the phases explicitly?

What is the end goal?

Reply:
The explicit thing we're adding is per-plugin information. So we can report "e2e plugin done, heptio-e2e plugin is still running.

Reply:
The "second phase" is the health information from the individual workers. That information like "this pod is alive, it hasn't given up" is definitely information we don't have right now: now we basically just have "it hasn't timed out yet" and "it's done."

Reply:
Just knowing the state of which plugins are running is useful UX, and a breadcrumb to call logs.